### PR TITLE
SQL Dependency for distributed caches

### PR DESCRIPTION
**Pull request by mohsinnasir**
Providing sample for keeping SQL database and distributed cache synchronized using SQL dependency

## Summary

Describe your changes here.

Fixes #Issue_Number (if available)

### CODE REVIEW

**Comment by mohsinnasir** [comment]
@stevestein Any update Regarding the PR

**Comment by mohsinnasir** [comment]
@stevestein Reminder Regarding the PR

**Comment by mohsinnasir** [comment]
@stevestein Kindly review the pull request . If there are any suggestions do tell me.

**Comment by rothja** [comment]
@mohsinnasir Thanks for the suggested edit. I'm reviewing it and will get back to you soon. #reassign:rothja

**Comment by rothja** [comment]
#assign:rothja

**Comment by mohsinnasir** [comment]
@rothja Thanks I ill be waiting and do let me know if there are any changes

**Comment by rothja** [comment]
@mohsinnasir Thanks for the ping on this one. My only concern on this section is that it points off to a third-party solution, which could be seen as a recommendation or preference for this solution. Do you know if this is a preferred solution by Microsoft (for example maybe we blogged about it or wrote about it on other sites)? If not, the way we typically do this is to refer to the type of solution and then say, ", such as <third-party solution>" and we would not make it its own H2. Let me know if you have any more details and I then I'll suggest an edit to get it synced with our documentation standards. Thanks!

**Comment by mohsinnasir** [comment]
@rothja Thanks for a quick response. However this section does not only points to thirdparty rather an architecture that is currently unavailable by the Microsoft distributed cache implementation. There are multiple distributed caches and Microsoft's Azure cache for redis does not have any feature to use SQL dependency for distributed caches. So, the documentation should have a place where a user knows there is a thirdparty which provides SQL dependency with distributed caches as well.

One more thing just like Mircosoft ASP.NET Core SignalR, it dosent have the support to provide backplane because it is only in memory so it mentions to provide SignalR backplane use Azure cache for redis or other thirdparties like mentioned [here](https://docs.microsoft.com/en-us/aspnet/signalr/overview/getting-started/introduction-to-signalr).

Built-in providers include:

Service Bus
SQL Server
Redis
Third-party providers include:

NCache.

**Comment by rothja** [comment]
@mohsinnasir Thanks for the explanation. I see how this could benefit readers to know about. Let me talk to the product team with this additional information, and I'll let you know what we recommend.

File: docs/framework/data/adonet/sql/sqldependency-in-an-aspnet-app.md [hunk 1]
```diff
@@ -50,7 +50,12 @@ The example in this section shows how to use <xref:System.Data.SqlClient.SqlDepe
...
   
  Now update the data in the database, using a Transact-SQL UPDATE command and refresh the page. The time displayed now indicates that the cache was refreshed with the new data from the database. Note that although the cache is updated, the time displayed on the page does not change until a postback event occurs.  
-  
+
+## Distributed cache synchronization using SQL Dependency
+Some of the third-party distributed caches like [NCache](https://www.alachisoft.com/ncache/) also provides support to keep SQL database and cache synchronized using [SQL Depenpendcy](https://www.alachisoft.com/resources/docs/ncache/prog-guide/sql-dependency.html). Following is the sample implementation.
+
+- [Distributed Cache SQL Dependency Sample](https://github.com/Alachisoft/NCache-Samples/tree/master/dotnet/Dependencies/SQLDependency)
```
Comment (IEvangelist) [inline docs/framework/data/adonet/sql/sqldependency-in-an-aspnet-app.md:57]:
```suggestion
## Distributed cache synchronization using SQL Dependency

Some of the third-party distributed caches such as [NCache](https://www.alachisoft.com/ncache) provide support to synchronize the SQL database and cache using [SQL Dependency](https://www.alachisoft.com/resources/docs/ncache/prog-guide/sql-dependency.html). For more information and an example source code implementation, see [Distributed cache SQL Dependency sample](https://github.com/Alachisoft/NCache-Samples/tree/master/dotnet/Dependencies/SQLDependency).
```

**Review by IEvangelist — CHANGES_REQUESTED** [review]
Hi 👋 
Thank you so much for posting this issue. We really appreciate it. I have a few minor nits.

**Comment by mohsinnasir** [comment]
@IEvangelist Thanks for taking the time for reviewing it. I have reviewed the the suggestions and agree with them. Kindly review the changes and let me know if there are any other changes required.

**Review by IEvangelist — APPROVED** [review]
Looks good to me, we'll :shipit: - thank you 🙏

**Comment by IEvangelist** [comment]
Hey @rothja - noticed that you were waiting to hear back from the product team on this. What did you hear?

**Comment by rothja** [comment]
@IEvangelist David, yes I got approval from the product team to put this change in. Thank you, @mohsinnasir for your patience.

**Pull request merged**